gateway/cert_test.go

```diff
@@ -1608,6 +1611,28 @@ func TestStaticMTLSAPI(t *testing.T) {
 		return ts, clientCertID, clientCert
 	}
 
+	generatePublicKey := func() []byte {
```

Comment:
It also seems like a utility function for internal/crypto, can have a unit test for it's particular output. Possibly needs a testing.TB argument to pin it's scope to tests (would we need this function in application scope? no)

Reply:
this is a test utility, do we need to move it to internal/crypto? 
I agree with testing.TB argument and it can live in testutils. wdyt?

Reply:
yes, move it, it's internal, it needs a unit test (as well as IsPublicKey also needs a unit test)